src/main/java/ro/secur/auth/exceptions/ResponseFactory.java

```diff
@@ -0,0 +1,17 @@
+package ro.secur.auth.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.time.LocalDateTime;
+
+public class ResponseFactory<T> {
+    /**
+     * @param ex         generic type of create
+     * @param httpStatus status code for 400 and 500 series
+     * @return responseEntity populated
+     */
+    ResponseEntity<Object> create(Exception ex, HttpStatus httpStatus) {
```

Comment:
The class is declared as generic ResponseEntity<T> but in the signature of this method you used Object. Please use generics here too.

Reply:
I need to return a type object to this method.